ui/src/main/java/edu/wpi/grip/ui/codeGeneration/java/TInput.java

```diff
@@ -0,0 +1,50 @@
+package edu.wpi.grip.ui.codegeneration.java;
+
+
+
+public class TInput extends TSocket {
+  private TOutput connectedOutput;
+  private String value;
+
+  public TInput(String type,String name,TOutput output){
+    super(type, name);
+    this.connectedOutput = output;
+  }
+
+  public TInput(String type,String name,String value){
+    super(type, name);
+    this.value = value;
+  }
+
+  public void setConnectedOutput(TOutput output){
+    this.value = null;
+    this.connectedOutput = output;
+  }
+
+  public String value(){
+    if(value!=null){
```

Comment:
Avoid using negation in if statements.  It would be clearer if this was written like this:
```
if (value == null) {
    return connectedOutput.name();
} else {
    return value;
}
```